manifests/plugin/puppet.pp

```diff
@@ -0,0 +1,30 @@
+# Pulp puppet plugin
+#
+# @param package_ensure The state or version to install
+# @param proxy_host Host of the proxy server.
+# @param proxy_port Port the proxy is running on.
+# @param proxy_username Proxy username for authentication.
+# @param proxy_password Proxy password for authentication.
+# @param wsgi_processes Number of WSGI processes to spawn for the puppet webapp
+class pulp::plugin::puppet (
+  $package_ensure = undef,
+  # Used in puppet_importer.json.erb
+  Optional[String] $proxy_host = $::pulp::proxy_url,
+  Optional[Integer[1, 65535]] $proxy_port = $::pulp::proxy_port,
+  Optional[String] $proxy_username = $::pulp::proxy_username,
+  Optional[String] $proxy_password = $::pulp::proxy_password,
+  # Used in apache.conf.erb
+  Integer $wsgi_processes = $::pulp::puppet_wsgi_processes,
+) {
+  selboolean { 'pulp_manage_puppet':
+    value      => 'on',
+    persistent => true,
+  }
```

Comment:
Maybe that's why it was an exec in the earlier code. I will admit I still have to test this on a real deployment.